controllers/preflightvalidationocp_reconciler.go

```diff
@@ -0,0 +1,271 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	openapivi "github.com/openshift/api/image/v1"
+	kmmv1beta1 "github.com/rh-ecosystem-edge/kernel-module-management/api/v1beta1"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/auth"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/filter"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/registry"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/statusupdater"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/syncronizedmap"
+)
+
+const (
+	driverToolkitSpecName        = "driver-toolkit"
+	driverToolkitJSONFilePath    = "etc/driver-toolkit-release.json"
+	releaseManifestImagesRefFile = "release-manifests/image-references"
+)
+
+type dtkRelease struct {
+	KernelVersion   string `json:"KERNEL_VERSION"`
+	RTKernelVersion string `json:"RT_KERNEL_VERSION"`
+	RHELVersion     string `json:"RHEL_VERSION"`
+}
+
+// PreflightValidationOCPReconciler reconciles a PreflightValidationOCP object
+type PreflightValidationOCPReconciler struct {
+	client             client.Client
+	filter             *filter.Filter
+	registry           registry.Registry
+	registryAuthGetter auth.RegistryAuthGetter
+	kernelOsDtkMapping syncronizedmap.KernelOsDtkMapping
+	statusUpdater      statusupdater.PreflightOCPStatusUpdater
+	scheme             *runtime.Scheme
+}
+
+func NewPreflightValidationOCPReconciler(
+	client client.Client,
+	filter *filter.Filter,
+	registry registry.Registry,
+	authFactory auth.RegistryAuthGetterFactory,
+	kernelOsDtkMapping syncronizedmap.KernelOsDtkMapping,
+	statusUpdater statusupdater.PreflightOCPStatusUpdater,
+	scheme *runtime.Scheme) *PreflightValidationOCPReconciler {
+	registryAuthGetter := authFactory.NewClusterAuthGetter()
+	return &PreflightValidationOCPReconciler{
+		client:             client,
+		filter:             filter,
+		registry:           registry,
+		registryAuthGetter: registryAuthGetter,
+		kernelOsDtkMapping: kernelOsDtkMapping,
+		statusUpdater:      statusUpdater,
+		scheme:             scheme,
+	}
+}
+
+func (r *PreflightValidationOCPReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("preflightvalidationocp").
+		For(&kmmv1beta1.PreflightValidationOCP{}, builder.WithPredicates(filter.PreflightOCPReconcilerUpdatePredicate())).
+		Watches(
+			&source.Kind{Type: &kmmv1beta1.PreflightValidation{}},
+			handler.EnqueueRequestsFromMapFunc(r.filter.EnqueueAllPreflightValidationsOCP),
+		).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: 1,
```

Comment:
This is the default value for the manager's respective global [option](https://github.com/kubernetes-sigs/controller-runtime/blob/92611cbdcaa9eb58f3e05697b20c0b282cb6447c/pkg/controller/controller.go#L39), but also the controller's default [option](https://github.com/kubernetes-sigs/controller-runtime/blob/92611cbdcaa9eb58f3e05697b20c0b282cb6447c/pkg/internal/controller/controller.go#L47). 

Do you emphasize here the fact that this reconciler should strictly stick with just one workqueue event processed at any time  (with its current implementation)? 

I have absolutely no problem with having this stated explicitly here, but I think increasing concurrency in any of the reconcilers would require very good verification that their implementation is "thread"-safe. So maybe we could skip it. Whatever you think is best. 

(mostly wrote this comment to make sure I understand all the aspects correctly)

Reply:
> I have absolutely no problem with having this stated explicitly here

I do 🙂 I think it might bring some confusion to someone reading this. "Why would they explicitly specify it? Are the defaults overridden anywhere?"

Reply:
I always like to state it explicitly, since not everybody remembers the default setting. It always helps to filter out debug versions that say: "maybe we have a concurrency issue here"

Reply:
As I mentioned above, I have no issue with it being explicit here, if we decide to leave it as is. IMHO:

1. having `MaxConcurrentReconciles=1` is the strict case for all KMM reconcilers currently
2. it's the other way round regarding concurrency and reconcilers, whoever tries to go for more than `MaxConcurrentReconciles > 1` would need to explicitly justify, verify every bit of code and unit-test extensively